enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,476 @@
+  a restart of the pod.
+
+##### Using LoadBalancer service type
+Common ways of exposing services outside of the cluster are using services of
+types NodePort or LoadBalancer. In MicroShift, ports from the service are
+picked up from ovnk (the CNI) and turned into iptables rules to forward them
+to the service IP. In this case, default router should be exposed using a
+LoadBalancer service type.
+
+ When using this kind of service ovnk will configure special iptables rules
+ that will forward traffic to the service IP instead. These rules are also
+ configured to use the node IP as destination address, as per
+ [LoadBalancer support](loadbalancer-service-support.md) feature.
+
+#### Firewalling ports
+Using LoadBalancer service will create special iptables rules with more
+precedence than those from firewalld. This means a `LoadBalancer` service is
+immune to such configurations. In order to firewall ports the user needs to
+take action by either creating `NetworkPolicy` resources or disabling the
+router.
+
+Disabling the router might be disruptive, as not all traffic may come from
+external sources. This procedure also requires a restart, which might not be
+desirable/possible in all situations.
+
+Using `NetworkPolicy` a user may control the traffic that is allowed into
+the cluster. By creating a deny-all policy the router will still remain
+operative for internal traffic, while rejecting all external traffic. An
+example follows:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: block-router
+  namespace: openshift-ingress
+spec:
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 10.42.0.0/16
+  podSelector:
+    matchLabels:
+      ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default
+  policyTypes:
+  - Ingress
+```
+Assuming the pod network is configured to use CIDR `10.42.0.0/16`, the above
+policy will block all traffic not coming from that network.
+
+Any of these methods is equivalent to using firewalld to block traffic, but
+using the CNI instead of a host level setting. This provides greater
+flexibility, as it is self contained within MicroShift APIs.
+
+If a user needs to block traffic to a port this needs a `NetworkPolicy`
+resource like the one shown above. This is NOT automatically done by
+MicroShift, user action is required.
+
+#### Exposing the router
+Relying on [LoadBalancer controller enhancement](loadbalancer-service-support.md)
+means the services are updated by this component. However, ovnk only turns IP
+addresses into iptables rules, it does not take Hostname, which is another
+valid field for the service status.
+
+Since the configuration uses NIC names and also host names, the controller
+needs to translate these to their corresponding IP addresses, and then
+configure them in the service. To keep up with the dynamic nature of IP
+addresses and names, it needs to perform periodic checks and updates.
```

Comment:
Should the check/update period be configurable?

Reply:
If its based on DNS lookup we could leverage on the TTL too. If it is based on files (like /etc/hosts) then you need to query again, which is where this parameter would come in handy. WDYT @DanielFroehlich @dhellmann?